### PR TITLE
SPORTSHUB-267 Recurring events backend implementation

### PR DESCRIPTION
**Pull request by brian2w**
JIRA link: https://owenyang.atlassian.net/browse/SPORTSHUB-267

## Why was your change needed?

Implementation of RecurringEvents feature.

## What was Changed?

Full implementation of RecurringEventsCron Cloud Run function.
Ported entire createEventWorkflow to Cloud Functions in Java.

Also changed MoveInactiveEvents to run at 12am Australia/Sydney timezone.

## Any extra notes for reviewers to know?

Write any extra notes you may have for the PR.
This may include specific code paths that need extra scrutiny, or how to review a specific portion of code.

### CODE REVIEW

File: frontend/services/src/recurringEvents/recurringEventsService.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    eventDataTemplate: eventData,
+    recurrenceData,
+  };
+  batch.set(docRef, recurringEventsData);
+
+  if (
+    isActive === recurringEventsStatus.Active &&
+    durationInDaysCeil(Timestamp.now(), recurrenceData.firstStartDate) <= recurrenceData.createDaysBefore
+  ) {
+    await createEvent(eventData, batch);
```
Comment (owya490) [inline frontend/services/src/recurringEvents/recurringEventsService.ts:33]: hmm we probably dont want the recurring event engine here right? we want the engine to be in functions and in a cloud cron job.

we want this function to be just the recurrence Template creator, and the actual event creation to still be delegated to the event service. 

No logic for the evaluation of creating an event should be done here. In other words, we don't want to check if we should create the first event here. (the first event will always be created regardless anyways).

Comment (brian2w) [reply frontend/services/src/recurringEvents/recurringEventsService.ts:33]: Ah yes, I agree we should always create first event regardless - this is not schedulling events

Comment (brian2w) [reply frontend/services/src/recurringEvents/recurringEventsService.ts:33]: Yep, the recurring event engine won't be here, but rather in firebase functions - this is still to do

File: functions/lib/move_inactive_events.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def get_and_move_private_inactive_events(today: date):
 
 
 
-@scheduler_fn.on_schedule(schedule="every day 00:00", region="australia-southeast1")
+@scheduler_fn.on_schedule(schedule="every day 00:00", region="australia-southeast1", time_zone="Australia/Sydney")
```
Comment (owya490) [inline functions/lib/move_inactive_events.py:73]: nice : D, we just need to make sure w deploy it - it might be worthwhile to split this into another PR so we don't lose track of this change

File: functions/lib/functions/src/main/java/com/functions/App.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Hello world!
+ *
+ */
+public class App 
+{
+    public static void main( String[] args )
+    {
+        System.out.println( "Hello World!" );
```
Comment (owya490) [inline functions/lib/functions/src/main/java/com/functions/App.java:11]: Lets change this to a Application Started or something along the lines of that

File: frontend/components/home/Hero.tsx
Comment (brian2w) [inline frontend/components/home/Hero.tsx:1]: Was reformatted via .pretterrc
